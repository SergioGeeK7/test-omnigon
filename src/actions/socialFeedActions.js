import {
  GET_DATA_SOCIAL_FEED,
  CLEAR_DATA_SOCIAL_FEED,
  TOGGLE_LOADING_INDICATOR_SOCIAL_FEED,
} from './types';
import { feedUrl, numerOfPost } from '../../config.json';

export const getSocialFeedData = page => (dispatch) => {
  const limit = numerOfPost * page;
  dispatch({
    type: TOGGLE_LOADING_INDICATOR_SOCIAL_FEED,
    payload: {
      isLoading: true,
    },
  });
  return fetch(`${feedUrl}?limit=${limit}`)
    .then(data => data.json())
    .then((data) => {
      dispatch({
        type: TOGGLE_LOADING_INDICATOR_SOCIAL_FEED,
        payload: {
          isLoading: false,
        },
      });
      return dispatch({
        type: GET_DATA_SOCIAL_FEED,
        payload: {
          data,
        },
      });
    });
};

export const refreshSocialFetch = () => (dispatch) => {
  dispatch({
    type: CLEAR_DATA_SOCIAL_FEED,
  });
  return getSocialFeedData(1)(dispatch);
};
