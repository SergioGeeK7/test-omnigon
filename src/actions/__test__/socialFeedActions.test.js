import mockStore from 'redux-mock-store';
import { getSocialFeedData, refreshSocialFetch } from '../socialFeedActions';

const store = mockStore({});
const response = '{}';

describe('socialFeedActions', () => {
  it('Get social feed data ACTION', () => {
    fetch.mockResponseSuccess(response);
    return store.dispatch(getSocialFeedData(1)).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  it('refresh social feed ACTION', () => {
    fetch.mockResponseSuccess(response);
    return store.dispatch(refreshSocialFetch()).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
