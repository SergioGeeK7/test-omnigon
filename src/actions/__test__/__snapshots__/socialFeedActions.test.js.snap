// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`socialFeedActions Get social feed data ACTION 1`] = `
Array [
  Object {
    "payload": Object {
      "isLoading": true,
    },
    "type": "TOGGLE_LOADING_INDICATOR_SOCIAL_FEED",
  },
  Object {
    "payload": Object {
      "isLoading": false,
    },
    "type": "TOGGLE_LOADING_INDICATOR_SOCIAL_FEED",
  },
  Object {
    "payload": Object {
      "data": Object {},
    },
    "type": "GET_DATA_SOCIAL_FEED",
  },
]
`;

exports[`socialFeedActions refresh social feed ACTION 1`] = `
Array [
  Object {
    "payload": Object {
      "isLoading": true,
    },
    "type": "TOGGLE_LOADING_INDICATOR_SOCIAL_FEED",
  },
  Object {
    "payload": Object {
      "isLoading": false,
    },
    "type": "TOGGLE_LOADING_INDICATOR_SOCIAL_FEED",
  },
  Object {
    "payload": Object {
      "data": Object {},
    },
    "type": "GET_DATA_SOCIAL_FEED",
  },
  Object {
    "type": "CLEAR_DATA_SOCIAL_FEED",
  },
  Object {
    "payload": Object {
      "isLoading": true,
    },
    "type": "TOGGLE_LOADING_INDICATOR_SOCIAL_FEED",
  },
  Object {
    "payload": Object {
      "isLoading": false,
    },
    "type": "TOGGLE_LOADING_INDICATOR_SOCIAL_FEED",
  },
  Object {
    "payload": Object {
      "data": Object {},
    },
    "type": "GET_DATA_SOCIAL_FEED",
  },
]
`;
